Build of socketio_mbed.c fails with warning

When using socketio_mbed.c it is failing with following warning:

mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c: In function ‘socketio_send’:                                                                                             /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c:334:33: error: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Werror=sign-compare]                                                                                                                                                                           334 |                 if (send_result != size)                                                                                                                                                  |                                 ^~                                                                                                                                                  /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c: In function ‘socketio_dowork’:                                                                                           /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c:388:33: error: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Werror=sign-compare]                                                                                                                                                                           388 |                 if (send_result != pending_socket_io->size)                                                                                                                               |                                 ^~                                                                                                                                                  /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c: In function ‘socketio_setoption’:                                                                                        /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c:451:43: error: unused parameter ‘socket_io’ [-Werror=unused-parameter]                                                      451 | int socketio_setoption(CONCRETE_IO_HANDLE socket_io, const char* optionName, const void* value)                                                                                           |                        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~                                                                                                                                 /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c:451:66: error: unused parameter ‘optionName’ [-Werror=unused-parameter]                                                     451 | int socketio_setoption(CONCRETE_IO_HANDLE socket_io, const char* optionName, const void* value)                                                                                           |                                                      ~~~~~~~~~~~~^~~~~~~~~~                                                                                                         /mnt/c/gitroot/azure-iot-sdk-c/c-utility/adapters/socketio_mbed.c:451:90: error: unused parameter ‘value’ [-Werror=unused-parameter]                                                          451 | int socketio_setoption(CONCRETE_IO_HANDLE socket_io, const char* optionName, const void* value)                                                                                           |                                                                              ~~~~~~~~~~~~^~~~~

--- a/adapters/socketio_mbed.c
+++ b/adapters/socketio_mbed.c
@@ -339,7 +339,7 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
             else
             {
                 int send_result = tcpsocketconnection_send(socket_io_instance->tcp_socket_connection, buffer, size);
-                if (send_result != size)
+                if ((size_t)send_result != size)
                 {
                     if (send_result < 0)
                     {
@@ -393,7 +393,7 @@ void socketio_dowork(CONCRETE_IO_HANDLE socket_io)
                 }
 
                 int send_result = tcpsocketconnection_send(socket_io_instance->tcp_socket_connection, (const char*)pending_socket_io->bytes, pending_socket_io->size);
-                if (send_result != pending_socket_io->size)
+                if ((size_t)send_result != pending_socket_io->size)
                 {
                     if (send_result < 0)
                     {
@@ -459,6 +459,9 @@ void socketio_dowork(CONCRETE_IO_HANDLE socket_io)
 int socketio_setoption(CONCRETE_IO_HANDLE socket_io, const char* optionName, const void* value)
 {
     /* Not implementing any options */
+    (void)socket_io;
+    (void)optionName;
+    (void)value;
     return MU_FAILURE;
 }
 

--- a/adapters/socketio_win32.c
+++ b/adapters/socketio_win32.c
@@ -564,8 +564,8 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
             }
             else
             {
-                int send_result = send(socket_io_instance->socket, (const char*)buffer, (int)size, 0);
-                if (send_result != (int)size)
+                ssize_t send_result = send(socket_io_instance->socket, (const char*)buffer, (int)size, 0);
+                if ((size_t)send_result != size)
                 {
                     int last_error = WSAGetLastError();
 
@@ -608,7 +608,6 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
 
 void socketio_dowork(CONCRETE_IO_HANDLE socket_io)
 {
-    int send_result;
     if (socket_io != NULL)
     {
         SOCKET_IO_INSTANCE* socket_io_instance = (SOCKET_IO_INSTANCE*)socket_io;
@@ -627,8 +626,8 @@ void socketio_dowork(CONCRETE_IO_HANDLE socket_io)
 
                 /* TODO: we need to do more than a cast here to be 100% clean
                 The following bug was filed: [WarnL4] socketio_win32 does not account for already sent bytes and there is a truncation of size from size_t to int */
-                send_result = send(socket_io_instance->socket, (const char*)pending_socket_io->bytes, (int)pending_socket_io->size, 0);
-                if (send_result != (int)pending_socket_io->size)
+                ssize_t send_result = send(socket_io_instance->socket, (const char*)pending_socket_io->bytes, (int)pending_socket_io->size, 0);
+                if ((size_t)send_result != pending_socket_io->size)
                 {
                     int last_error = WSAGetLastError();
                     if (last_error != WSAEWOULDBLOCK)

--- a/adapters/socketio_win32.c
+++ b/adapters/socketio_win32.c
@@ -564,7 +564,7 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
             }
             else
             {
-                ssize_t send_result = send(socket_io_instance->socket, (const char*)buffer, (int)size, 0);
+                int send_result = send(socket_io_instance->socket, (const char*)buffer, (int)size, 0);
                 if ((size_t)send_result != size)
                 {
                     int last_error = WSAGetLastError();
@@ -626,7 +626,7 @@ void socketio_dowork(CONCRETE_IO_HANDLE socket_io)
 
                 /* TODO: we need to do more than a cast here to be 100% clean
                 The following bug was filed: [WarnL4] socketio_win32 does not account for already sent bytes and there is a truncation of size from size_t to int */
-                ssize_t send_result = send(socket_io_instance->socket, (const char*)pending_socket_io->bytes, (int)pending_socket_io->size, 0);
+                int send_result = send(socket_io_instance->socket, (const char*)pending_socket_io->bytes, (int)pending_socket_io->size, 0);
                 if ((size_t)send_result != pending_socket_io->size)
                 {
                     int last_error = WSAGetLastError();

